iwyu.cc

```diff
@@ -3110,9 +3113,53 @@ class InstantiatedTemplateVisitor
     // We attribute all uses in an instantiated template to the
     // template's caller.
     ReportTypeUse(caller_loc(), actual_type);
+
+    // Report also all previous explicit instantiations (declarations and
+    // defintions) as uses of the caller's location.
+    ReportExplicitInstantiations(actual_type);
+
     return Base::VisitSubstTemplateTypeParmType(type);
   }
 
+  bool VisitTemplateSpecializationType(
+      clang::TemplateSpecializationType* type) {
```

Comment:
`TemplateSpecializationType` has been pulled into the namespace with `using` at the top of the file. If you remove the namespace specifier, the declaration fits on one line.